packages/control-property-editor/src/panels/changes/LongText.tsx

```diff
@@ -0,0 +1,110 @@
+            let totalCharInLine1 = 0;
+            let totalCharInLine2 = 0;
+            const midpoint = Math.floor(text.length / 2);
+            // Find the number of characters that fit in the first line
+            let part1done = false;
+            let part2done = false;
+            virtualDiv1.textContent = text;
+            // If the text fits in one line, return it without ellipses
+            if (virtualDiv1.scrollHeight <= lineHeight) {
+                document.body.removeChild(virtualDiv1);
+                document.body.removeChild(virtualDiv2);
+                return { firstLine: text, secondLine: '' };
+            }
+            virtualDiv1.textContent = '';
+            for (let i = 1, j = text.length - 1; i <= midpoint + 1 || j > midpoint + 1; i++, j--) {
+                if (part1done && part2done) {
+                    break;
+                }
+                if (i <= midpoint + 1 && !part1done) {
+                    virtualDiv1.textContent = text.slice(0, i);
+                    if (virtualDiv1.scrollHeight > lineHeight) {
+                        part1done = true;
+                    }
+                    totalCharInLine1 += 1;
+                }
+                if (j > midpoint && !part2done) {
+                    virtualDiv2.textContent = text.slice(j - text.length);
+                    if (virtualDiv2.scrollHeight > lineHeight) {
+                        part2done = true;
+                    }
+                    totalCharInLine2 += 1;
+                }
+            }
+
+            document.body.removeChild(virtualDiv1);
+            document.body.removeChild(virtualDiv2);
+
+            if (!part1done && !part2done) {
+                return {
+                    firstLine: virtualDiv1.textContent,
+                    secondLine: virtualDiv2.textContent ?? ''
+                };
+            }
+            const totalTextLength = text.length - labelLength;
+            const totalCharsThatFitInTwoLines = totalCharInLine1 + totalCharInLine2 - labelLength;
+            const charsToTruncate = text.length - labelLength - totalCharsThatFitInTwoLines + 3 + 3; // 3 for ellipses and 3 for Math.floor
+
+            // Calculate the center and remove characters proportionally
+            const center = Math.floor(totalTextLength / 2);
+            const firstPart =
+                text.slice(0, labelLength) + text.slice(labelLength, center - Math.floor(charsToTruncate / 2));
+            const secondPart = text.slice(center + Math.floor(charsToTruncate / 2));
+            const truncatedText = firstPart + '...' + secondPart;
+
+            return {
+                firstLine: truncatedText.slice(0, totalCharInLine1 - 1).trim(),
+                secondLine: truncatedText.slice(totalCharInLine1 - 1).trim()
+            };
+        };
+
+        const { firstLine, secondLine } = truncateText(finalText);
+        setFirstLine(firstLine);
+        setSecondLine(secondLine);
+    }, [longText, label]);
```

Comment:
This does not seem like the right approach and has potential performance impact, because each time `scrollHeight` forces layout/reflow which is expensive and we do it for each character.

Reply:
removed. add implemented as we discussed.